core/src/com/unciv/logic/city/PopulationManager.kt

```diff
@@ -109,34 +112,49 @@ class PopulationManager {
         addPopulation(-population + count)
     }
 
-    internal fun autoAssignPopulation(foodWeight: Float = 1f) {
+    internal fun autoAssignPopulation() {
+        cityInfo.cityStats.update()  // calculate current stats with current assignments
+        cityInfo.currentGPPBonus = cityInfo.getGreatPersonPercentageBonus()  // pre-calculate
+        var specialistFoodBonus = 2f  // See CityStats.calcFoodEaten()
+        for (unique in cityInfo.getMatchingUniques(UniqueType.FoodConsumptionBySpecialists))
+            if (cityInfo.matchesFilter(unique.params[1]))
+                specialistFoodBonus *= unique.params[0].toPercent()
+        specialistFoodBonus = 2f - specialistFoodBonus
+        
         for (i in 1..getFreePopulation()) {
             //evaluate tiles
-            val bestTile: TileInfo? = cityInfo.getTiles()
+            val (bestTile, valueBestTile) = cityInfo.getTiles()
                     .filter { it.aerialDistanceTo(cityInfo.getCenterTile()) <= 3 }
                     .filterNot { it.providesYield() }
-                    .maxByOrNull { Automation.rankTileForCityWork(it, cityInfo, foodWeight) }
-            val valueBestTile = if (bestTile == null) 0f
-            else Automation.rankTileForCityWork(bestTile, cityInfo, foodWeight)
-
-            val bestJob: String? = getMaxSpecialists()
+                    .associateWith { Automation.rankTileForCityWork(it, cityInfo) }
+                    .maxByOrNull { it.value }
+                    ?: object : Map.Entry<TileInfo?, Float> {
+                        override val key: TileInfo? = null
+                        override val value = 0f
+                    }
+
+            val bestJob: String? = if (cityInfo.manualSpecialists) null else getMaxSpecialists()
                     .filter { specialistAllocations[it.key]!! < it.value }
                     .map { it.key }
-                    .maxByOrNull { Automation.rankSpecialist(getStatsOfSpecialist(it), cityInfo) }
-
+                    .maxByOrNull { Automation.rankSpecialist(it, cityInfo) }
 
             var valueBestSpecialist = 0f
             if (bestJob != null) {
-                val specialistStats = getStatsOfSpecialist(bestJob)
-                valueBestSpecialist = Automation.rankSpecialist(specialistStats, cityInfo)
+                valueBestSpecialist = Automation.rankSpecialist(bestJob, cityInfo)
             }
 
             //assign population
             if (valueBestTile > valueBestSpecialist) {
-                if (bestTile != null)
+                if (bestTile != null) {
                     cityInfo.workedTiles = cityInfo.workedTiles.withItem(bestTile.position)
-            } else if (bestJob != null) specialistAllocations.add(bestJob, 1)
+                    cityInfo.cityStats.currentCityStats[Stat.Food] += bestTile.getTileStats(cityInfo, cityInfo.civInfo)[Stat.Food]
```

Comment:
It should run and clean up my temp use of it at the end when it runs the update(). I opted to just reuse this instead of copying to a Transient Stats variable in CityInfo.

The only stat that the Automation needs is Food. Replicating all of the stats if a lot of redundant code. I could run the update() after each Citizen is assigned, but that doubled the runtime

Reply:
I can do the change later tonight

Reply:
Done